stdlib/2.7/getopt.pyi

```diff
@@ -0,0 +1,27 @@
+from typing import List, Tuple
+
+class GetoptError(Exception):
+    opt = ... # type: str
+    msg = ... # type: str
+    def __init__(self, msg: str, opt: str=...) -> None: ...
+    def __str__(self) -> str: ...
+
+error = GetoptError
+
+def getopt(args: List[str], shortopts: str,
+           longopts: List[str]=...) -> Tuple[List[Tuple[str, str]],
+                                             List[str]]: ...
+
+def gnu_getopt(args: List[str], shortopts: str,
+           longopts: List[str]=...) -> Tuple[List[Tuple[str,str]],
+                                             List[str]]: ...
+
+def do_longs(opts: List[Tuple[str,str]], opt: str,
```

Comment:
Theres no rason to insult the documentation. Anyway, please omit the
undocumented functions in this case. The public/private distinction is
often a pretty bad signal.

On Tuesday, May 24, 2016, Dakkaron <notifications@github.com> wrote:

> In stdlib/2.7/getopt.pyi
> <https://github.com/python/typeshed/pull/221#discussion_r64515582>:
>
> > +    opt = ... # type: str
> > +    msg = ... # type: str
> > +    def __init__(self, msg: str, opt: str=...) -> None: ...
> > +    def __str__(self) -> str: ...
> > +
> > +error = GetoptError
> > +
> > +def getopt(args: List[str], shortopts: str,
> > +           longopts: List[str]=...) -> Tuple[List[Tuple[str, str]],
> > +                                             List[str]]: ...
> > +
> > +def gnu_getopt(args: List[str], shortopts: str,
> > +           longopts: List[str]=...) -> Tuple[List[Tuple[str,str]],
> > +                                             List[str]]: ...
> > +
> > +def do_longs(opts: List[Tuple[str,str]], opt: str,
>
> I don't know, I just added everything that was in the source. Those
> functions are basically only helper functions for the "real" functions of
> that module, but they are not marked as private. I have to admit, I rarely
> have a look into the documentation due to it's low quality. Do you only
> want documented functions or do you want all non-private functions?
>
> —
> You are receiving this because you commented.
> Reply to this email directly or view it on GitHub
> <https://github.com/python/typeshed/pull/221/files/bf7c1c2d38e4bba00f5ee3c1e9e1f03963e77c4b#r64515582>
>


-- 
--Guido (mobile)